dcpy/lifecycle/distribute/socrata.py

```diff
@@ -48,55 +48,52 @@ def _dist_from_local(
         "--ignore-validation-errors",
         help="Ignore Validation Errors? Will still perform validation, but ignore errors, allowing a push",
     ),
+    skip_validation: bool = typer.Option(
+        False,
+        "-y",  # -y(olo)
+        "--skip-validation",
+        help="Skip Validation Altogether",
+    ),
 ):
     md = m.Metadata.from_yaml(metadata_path or package_path / "metadata.yml")
     dest = md.get_destination(dataset_destination_id)
     assert dest.type == "socrata"
 
-    logger.info("Validating package")
-    try:
-        errors = v.validate_package(package_path, md)
-    except Exception as e:
-        errors = [str(e)]
+    if not skip_validation:
+        logger.info("Validating package")
+        validation = v.validate_package(package_path, md)
+        errors = validation.get_dataset_errors()
 
-    if len(errors) > 0:
-        error_msg = f"Errors Found! {errors}"
-        if ignore_validation_errors:
-            logger.warn(error_msg)
-        else:
-            logger.error(error_msg)
-            raise Exception(error_msg)
+        if len(errors) > 0:
+            if ignore_validation_errors:
+                logger.warn("Errors Found! But continuing to distribute")
+                validation.pretty_print_errors()
+            else:
+                error_msg = "Errors Found! Aborting distribute"
+                logger.error(error_msg)
+                validation.pretty_print_errors()
+                raise Exception(error_msg)
 
     ds_name_to_push = dest.datasets[0]  # socrata will only have one dataset
 
-    match md.dataset_package.get_dataset(ds_name_to_push).type:
+    match md.package.get_dataset(ds_name_to_push).type:
         case "shapefile":
             soc_pub.push_shp(md, dataset_destination_id, package_path, publish=publish)
-
         case _:
             # TODO
             raise Exception("Only shapefiles have been implemented so far")
 
 
-@app.command("from_s3")
+@socrata_app.command("from_s3")
 def _dist_from_s3(
-    product_name: str = typer.Option(
-        None,
-        "-n",
-        "--name",
-        help="product name",
-    ),
-    version: str = typer.Option(
-        None,
-        "-v",
-        "--version",
-        help="package version",
-    ),
-    dataset_destination_id: str = typer.Option(
+    product_name: str,
+    version: str,
+    dataset_destination_id: str,
+    dataset: str = typer.Option(
         None,
-        "-d",
-        "--dest",
-        help="Dataset Destination Id",
+        "-d",  # d is already taken, unfortunately
```

Comment:
but this still uses `-d`?

Reply:
ha, that was a tongue-in-cheek comment that become unnecessary. Will fix ✂️